tc/sgx/trusted_worker_manager/enclave/kme/CMakeLists.txt

```diff
@@ -0,0 +1,76 @@
+# Copyright 2020 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+CMAKE_MINIMUM_REQUIRED(VERSION 3.2 FATAL_ERROR)
+PROJECT(libavalon-kme-enclave  CXX)
+
+INCLUDE(../CMakeVariables.txt)
+
+# XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+# ADDITIONAL LIBRARIES
+# XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+
+SET(TCF_TOP_DIR "${CMAKE_CURRENT_SOURCE_DIR}/../../../../../")
+SET(TCF_CORE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/../../../")
+#SET(WPE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/wpe)
+
+# XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+
+FILE(GLOB PROJECT_HEADERS *.h ../*.h)
+FILE(GLOB PROJECT_SOURCES *.cpp *.c ../*.cpp)
+FILE(GLOB PROJECT_EDL kme_enclave.edl)
+FILE(GLOB PROJECT_CONFIG *.xml)
+SOURCE_GROUP("Source" FILES ${PROJECT_HEADERS} ${PROJECT_SOURCES} ${PROJECT_EDL})
+
+SET(EDL_PATH ${CMAKE_CURRENT_SOURCE_DIR})
+
+SGX_EDGE_TRUSTED(${PROJECT_EDL} ${EDL_PATH} PROJECT_EDGE_SOURCES)
+
+SOURCE_GROUP("Generated" FILES ${PROJECT_EDGE_SOURCES})
+
+# XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+# Compile targets
+# XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+
+# CMake likes to prepend "lib" to project name when creating library...since we
+# already have lib in the project name, liblibavalon-singleton-enclave is stupid.  Stop
+# CMake from doing it.
+SET(CMAKE_SHARED_LIBRARY_PREFIX "")
+PROJECT(libavalon-kme-enclave C CXX)
+
+ADD_LIBRARY(${PROJECT_NAME} SHARED ${PROJECT_HEADERS} ${PROJECT_SOURCES} ${PROJECT_EDGE_SOURCES} ${PROJECT_EDL})
+SGX_SIGN_ENCLAVE(${PROJECT_NAME} ${TCF_ENCLAVE_CODE_SIGN_PEM} ${PROJECT_CONFIG})
```

Comment:
Above 2 lines are common for all enclave types. Can this also be moved to CMakeVariables.txt

Reply:
These are just function calls whose implementation is already part of CMakeVariables.txt. I think, keeping it here makes better readability.